App Crashes When Playing A Video Podcast Episode

### Checklist

- [x] I have used the search function for [**OPEN**](https://github.com/AntennaPod/AntennaPod/issues) issues to see if someone else has already submitted the same bug report.
- [X] I have **also** used the search function for [**CLOSED**](https://github.com/AntennaPod/AntennaPod/issues?q=is%3Aissue+is%3Aclosed) issues to see if the problem is already solved and just waiting to be released.
- [X] I will describe the problem with as much detail as possible.
- [X] If the bug only to occurs with a certain podcast, I will include the URL of that podcast.

### App version

3.4.0 5f5d744e7

### Where did you get the app from

Google Play

### Android version

LineageOS 13 (lineage_raven-userdebug 13 TQ3A.230901.001)

### Device model

Google Pixel 6 Pro

### First occurred

1 or 2 days ago

### Steps to reproduce

1. Press play on a video podcast episode

### Expected behaviour

Play the video podcast in the in-app video player

### Current behaviour

As soon as I hit the play button, the video interface pops up, but then the app crashes completely, without warning

### Logs

## Environment
Android version: 13
OS version: 5.10.157-gd6584346874f
AntennaPod version: 3.4.0
Model: Pixel 6 Pro
Device: raven
Product: raven

## Crash info
Time: 29-05-2024 09:26:31
AntennaPod version: 3.4.0

## StackTrace
```
java.lang.NullPointerException: Attempt to invoke virtual method 'java.util.List de.danoeh.antennapod.playback.service.internal.ExoPlayerWrapper.getAudioTracks()' on a null object reference
	at de.danoeh.antennapod.playback.service.internal.LocalPSMP.getAudioTracks(SourceFile:582)
	at de.danoeh.antennapod.playback.service.PlaybackService.getAudioTracks(SourceFile:1772)
	at de.danoeh.antennapod.playback.service.PlaybackController.getAudioTracks(SourceFile:437)
	at de.danoeh.antennapod.ui.screen.playback.video.VideoplayerActivity.onPrepareOptionsMenu(SourceFile:570)
	at android.app.Activity.onPreparePanel(Activity.java:4398)
	at androidx.activity.ComponentActivity.onPreparePanel(SourceFile:497)
	at androidx.appcompat.view.WindowCallbackWrapper.onPreparePanel(SourceFile:100)
	at androidx.appcompat.app.AppCompatDelegateImpl$AppCompatWindowCallback.onPreparePanel(SourceFile:3170)
	at androidx.appcompat.app.AppCompatDelegateImpl.preparePanel(SourceFile:1940)
	at androidx.appcompat.app.AppCompatDelegateImpl.doInvalidatePanelMenu(SourceFile:2194)
	at androidx.appcompat.app.AppCompatDelegateImpl$2.run(SourceFile:276)
	at android.os.Handler.handleCallback(Handler.java:942)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:201)
	at android.os.Looper.loop(Looper.java:288)
	at android.app.ActivityThread.main(ActivityThread.java:7924)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
```

I'm having this same issue with all podcasts, video or no video. Exactly as described above. 

Android 14
Google Pixel 7 Pro
AntennaPod 3.4.0 (F-droid)

```
## Environment
Android version: 14
OS version: 5.10.189-android13-4-00012-g1217bb583cc5-ab11174560
AntennaPod version: 3.4.0
Model: Pixel 7 Pro
Device: cheetah
Product: cheetah

## Crash info
Time: 29-05-2024 13:37:55
AntennaPod version: 3.4.0

## StackTrace
```
java.lang.RuntimeException: Unable to start service de.danoeh.antennapod.playback.service.PlaybackService@fb53169 with Intent { cmp=de.danoeh.antennapod/.playback.service.PlaybackService (has extras) }: android.app.ForegroundServiceStartNotAllowedException: Service.startForeground() not allowed due to mAllowStartForeground false: service de.danoeh.antennapod/.playback.service.PlaybackService
	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:5053)
	at android.app.ActivityThread.-$$Nest$mhandleServiceArgs(Unknown Source:0)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2390)
	at android.os.Handler.dispatchMessage(Handler.java:107)
	at android.os.Looper.loopOnce(Looper.java:232)
	at android.os.Looper.loop(Looper.java:317)
	at android.app.ActivityThread.main(ActivityThread.java:8501)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:878)
Caused by: android.app.ForegroundServiceStartNotAllowedException: Service.startForeground() not allowed due to mAllowStartForeground false: service de.danoeh.antennapod/.playback.service.PlaybackService
	at android.app.ForegroundServiceStartNotAllowedException$1.createFromParcel(ForegroundServiceStartNotAllowedException.java:54)
	at android.app.ForegroundServiceStartNotAllowedException$1.createFromParcel(ForegroundServiceStartNotAllowedException.java:50)
	at android.os.Parcel.readParcelableInternal(Parcel.java:5015)
	at android.os.Parcel.readParcelable(Parcel.java:4997)
	at android.os.Parcel.createExceptionOrNull(Parcel.java:3177)
	at android.os.Parcel.createException(Parcel.java:3166)
	at android.os.Parcel.readException(Parcel.java:3149)
	at android.os.Parcel.readException(Parcel.java:3091)
	at android.app.IActivityManager$Stub$Proxy.setServiceForeground(IActivityManager.java:6861)
	at android.app.Service.startForeground(Service.java:775)
	at de.danoeh.antennapod.playback.service.internal.PlaybackServiceStateManager.startForeground(SourceFile:22)
	at de.danoeh.antennapod.playback.service.PlaybackService.onStartCommand(SourceFile:500)
	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:5035)
	... 9 more
```

> I'm having this same issue with all podcasts, video or no video. Exactly as described above.
> 
> Android 14 Google Pixel 7 Pro AntennaPod 3.4.0 (F-droid)
> 
> ```
> ## Environment
> Android version: 14
> OS version: 5.10.189-android13-4-00012-g1217bb583cc5-ab11174560
> AntennaPod version: 3.4.0
> Model: Pixel 7 Pro
> Device: cheetah
> Product: cheetah
> 
> ## Crash info
> Time: 29-05-2024 13:37:55
> AntennaPod version: 3.4.0
> ```
> 
> ```
> ## StackTrace
> java.lang.RuntimeException: Unable to start service de.danoeh.antennapod.playback.service.PlaybackService@fb53169 with Intent { cmp=de.danoeh.antennapod/.playback.service.PlaybackService (has extras) }: android.app.ForegroundServiceStartNotAllowedException: Service.startForeground() not allowed due to mAllowStartForeground false: service de.danoeh.antennapod/.playback.service.PlaybackService
> 	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:5053)
> 	at android.app.ActivityThread.-$$Nest$mhandleServiceArgs(Unknown Source:0)
> 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2390)
> 	at android.os.Handler.dispatchMessage(Handler.java:107)
> 	at android.os.Looper.loopOnce(Looper.java:232)
> 	at android.os.Looper.loop(Looper.java:317)
> 	at android.app.ActivityThread.main(ActivityThread.java:8501)
> 	at java.lang.reflect.Method.invoke(Native Method)
> 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
> 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:878)
> Caused by: android.app.ForegroundServiceStartNotAllowedException: Service.startForeground() not allowed due to mAllowStartForeground false: service de.danoeh.antennapod/.playback.service.PlaybackService
> 	at android.app.ForegroundServiceStartNotAllowedException$1.createFromParcel(ForegroundServiceStartNotAllowedException.java:54)
> 	at android.app.ForegroundServiceStartNotAllowedException$1.createFromParcel(ForegroundServiceStartNotAllowedException.java:50)
> 	at android.os.Parcel.readParcelableInternal(Parcel.java:5015)
> 	at android.os.Parcel.readParcelable(Parcel.java:4997)
> 	at android.os.Parcel.createExceptionOrNull(Parcel.java:3177)
> 	at android.os.Parcel.createException(Parcel.java:3166)
> 	at android.os.Parcel.readException(Parcel.java:3149)
> 	at android.os.Parcel.readException(Parcel.java:3091)
> 	at android.app.IActivityManager$Stub$Proxy.setServiceForeground(IActivityManager.java:6861)
> 	at android.app.Service.startForeground(Service.java:775)
> 	at de.danoeh.antennapod.playback.service.internal.PlaybackServiceStateManager.startForeground(SourceFile:22)
> 	at de.danoeh.antennapod.playback.service.PlaybackService.onStartCommand(SourceFile:500)
> 	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:5035)
> 	... 9 more
> ```

Looks like what you found can be easily fixed using the following advice:

https://stackoverflow.com/questions/77786169/service-startforeground-not-allowed-due-to-mallowstartforeground-false-when-tr

I don't think it's the same as what I found though

--- a/playback/service/src/main/java/de/danoeh/antennapod/playback/service/internal/LocalPSMP.java
+++ b/playback/service/src/main/java/de/danoeh/antennapod/playback/service/internal/LocalPSMP.java
@@ -31,6 +31,7 @@ import org.greenrobot.eventbus.EventBus;
 
 import java.io.File;
 import java.io.IOException;
+import java.util.Collections;
 import java.util.List;
 import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.TimeUnit;
@@ -579,6 +580,9 @@ public class LocalPSMP extends PlaybackServiceMediaPlayer {
     }
 
     public List<String> getAudioTracks() {
+        if (mediaPlayer == null) {
+            return Collections.emptyList();
+        }
         return mediaPlayer.getAudioTracks();
     }
 
@@ -587,6 +591,9 @@ public class LocalPSMP extends PlaybackServiceMediaPlayer {
     }
 
     public int getSelectedAudioTrack() {
+        if (mediaPlayer == null) {
+            return -1;
+        }
         return mediaPlayer.getSelectedAudioTrack();
     }
 

